### ISSUE TITLE
Question about dependencies

### ISSUE THREAD

**Issue by akhmerov** [ExecutableBookProject/sphinx-jupyter-book-theme#4]
How does this theme depend on IPython, nbsphinx, and ipywidgets? These don't appear to be used anywhere in the repo aside of setup.py.

**Comment by rossbar** [comment]
Re: IPython - it is likely not a dependency of the theme itself, but is nice for testing code blocks as it provides the pygments syntax highlighter (which suppresses a slew of sphinx errors).

I wouldn't be surprised if the other dependencies are there for similar reasons. Breaking them out into an `extras_require` entry in the `setup.py` would be good.

**Comment by akhmerov** [comment]
Thanks for the clarification. If it's only for testing, there's also `tests_require`.

**Comment by choldgraf** [comment]
Yep - that's definitely correct. More generally, this repository is very much in an "this is super young and there has been almost effort made into professionalizing things yet" phase :-)

I'd be +1 on any changes that improve this so it's less confusing or bulky.
